### PR TITLE
Prototype buildpack that performs vuls scan

### PR DESCRIPTION
**Pull request by ohsh6o**
This is a prototype buildpack that will send scan data to a [vuls](https://vuls.io/) server for processing vulnerable dependencies.

### CODE REVIEW

File: bin/supply [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+echo "Installing the vuls to ${VULS_DIR}"
+
+mkdir -p $VULS_DIR
+mkdir -p $BUILD_DIR/.profile.d
+
+cp -r $ROOT_DIR/lib/dist $VULS_DIR/
+
+echo "Download vuls asset from ${VULS_URL}"
+curl -L -o $VULS_DIR/vuls.tar.gz $VULS_URL
```
Comment (mike-stern) [inline bin/supply]: pull `vuls.tar.gz` into a variable as it is repeated on L25 & L27 below

File: bin/supply [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+echo "-----> VulsBuildpack/supply"
+
+BIN_DIR=$(cd $(dirname $0); pwd)
+ROOT_DIR=$(dirname $BIN_DIR)
+BUILD_DIR=$1
+CACHE_DIR=$2
+DEPS_DIR=$3
+INDEX=$4
+
+VULS_URL="${VULS_URL:-https://github.com/future-architect/vuls/releases/download/v0.9.6/vuls_0.9.6_linux_amd64.tar.gz}"
```
Comment (mike-stern) [inline bin/supply]: make `0.9.6` an externalizable variable at some point in the future

File: bin/detect [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+#!/usr/bin/env bash
+
+echo "-----> VulsBuildpack/detect"
```
Comment (mike-stern) [inline bin/detect:3]: at some point does this become a heuristic scan of deploy targe for known types or does the buildpack plan to be forever explicitly invoked?

**Review by mike-stern — APPROVED** [review]

File: bin/supply (hunk 2 above)
Comment (ohsh6o) [reply bin/supply]: So the point of the `VULS_URL="${VULS_URL:-https://sensible-default/path/to/download}` is actually to externalize the whole URL. I found out from experimenting with using the `latest` tag that this does not work well. One of my issues now is references another variable to the right of the `:-` and it being weird, but I can definitely try that in a bit.

File: bin/detect (hunk 3 above)
Comment (ohsh6o) [reply bin/detect:3]: I had read some recommendations that for meta-buildpack and multi-buildpack style that is opt-in only, advisable to code it that way. That being said, are you asking if we should build-in a detection pass that recursively looks in the staged buildpack files for a support dep file?

Before, I would have thought this possible. It was your link to the Brett Mogilefsky presentation at the CF2018 Summit and our follow-up discussion Tuesday that makes me think that might not work. In short: our buildpack has to exist before the others do, and stages a hook file into others later to be seen, but it does not see at the detect lifecycle phase when `bin/detect` is called any of the Python files from our 10x-dux-app for example, and especially the `Pipfile` it needs to look for.

Comment (ohsh6o) [reply bin/detect:3]: Also, this review comment was useful in a unrelated way: you clued me into the double shebang.

File: bin/supply (hunk 2 above)
Comment (ohsh6o) [reply bin/supply]: I parameterized it anyway.

File: bin/supply (hunk 1 above)
Comment (ohsh6o) [reply bin/supply]: Sounds like a good plan, will do.

Comment (ohsh6o) [reply bin/supply]: Added.

File: bin/detect (hunk 3 above)
Comment (ohsh6o) [reply bin/detect:3]: I am not going to close this yet, but I am opening up PR for review. I think we can wait for Bret's viewpoint and decide if we can do this, and prioritize.

**Review by mike-stern — APPROVED** [review]
As it stands this is ready for merge. The comment on 'detect' is an open question, but not something to hold the merge up.

**Comment by ohsh6o** [comment]
> As it stands this is ready for merge. The comment on 'detect' is an open question, but not something to hold the merge up.

Sorry for the delayed response. I will merge this for now. We will review the detect option at a later time.

**Pull request merged**